avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/RecordVisitor.java

```diff
@@ -49,14 +74,19 @@ public Schema builtAvroSchema() {
     @Override
     public void property(BeanProperty writer) throws JsonMappingException
     {
-        Schema schema = schemaForWriter(writer);
-        _fields.add(new Schema.Field(writer.getName(), schema, null, null));
+        if (_overridden) {
+            return;
+        }
+        _fields.add(schemaFieldForWriter(writer, false));
     }
 
     @Override
     public void property(String name, JsonFormatVisitable handler,
             JavaType type) throws JsonMappingException
     {
+        if (_overridden) {
```

Comment:
Yeah; The idea was if the developer specified `@AvroSchema`, we just ignore everything about the POJO that we're visiting (because we'd be discarding it anyways).